api/common.go

```diff
@@ -147,3 +151,27 @@ func LoadOrCreateTrustKey(trustKeyPath string) (libtrust.PrivateKey, error) {
 	}
 	return trustKey, nil
 }
+
+// VerifyTarballNotImage verifies that the given tarball is not an image tarball.
+func VerifyTarballNotImage(reader io.Reader) (*bytes.Buffer, error) {
+	decomArchive, err := archive.DecompressStream(reader)
+	if err != nil {
+		return nil, err
+	}
+	data := bytes.NewBuffer(nil)
+	r := io.TeeReader(decomArchive, data)
+	tr := tar.NewReader(r)
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return data, err
+		}
+		if hdr.Name == "repositories" {
+			return data, fmt.Errorf("tarball must not be an image: detected \"repositories\" file")
+		}
+	}
+	return data, nil
+}
```

Comment:
This will buffer the whole tarball in memory. I don't think that's okay because some images may be too large to fit in memory.